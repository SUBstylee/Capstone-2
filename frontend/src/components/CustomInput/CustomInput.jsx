import './CustomInput.scss';

const CustomInput = ({labelText,id, ...otherProps}) => {
    const labelTextArr=labelText.split('');
    let labelKey=0;

    return (
        <div className='CustomInput'>
            <input {...otherProps} id={id}/>
            <label htmlFor={id}>{
                labelTextArr.map((char)=><span style={{transitionDelay:`${labelKey*50}ms`}} key={labelKey++}>{char}</span>)
            }</label>
        </div>
    );
};

export default CustomInput;