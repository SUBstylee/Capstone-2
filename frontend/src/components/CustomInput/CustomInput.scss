.CustomInput{
    position: relative;
    margin: 20px 0 40px;
    width: 300px;
    input{
        background-color: #fff !important;
        border: none;
        border-bottom: 2px #919191 solid;
        display: block;
        width: 100%;
        padding: 15px 0;
        font-size: 20px;
    };

    input:focus, input:valid{
        outline: 0;
        border-bottom-color: rgb(28, 28, 165);
    };

    input:-webkit-autofill,
    input:-webkit-autofill:hover, 
    input:-webkit-autofill:focus, 
    input:-webkit-autofill:active{
        -webkit-box-shadow: 0 0 0 30px white inset !important;
    }

    label{
        position: absolute;
        top:15px;
        left: 0;
    }
    label span{
        display: inline-block;
        font-size: 18px;
        min-width: 5px;
        transition: 0.3s cubic-bezier(0.68,-0.55,0.265,1.55);
    }
    input:focus + label span,
    input:valid + label span{
        color: rgb(28, 28, 165);
        transform: translateY(-30px);
    }
};