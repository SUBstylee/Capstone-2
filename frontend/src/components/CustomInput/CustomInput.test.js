import { render } from '@testing-library/react';
import CustomInput from './CustomInput';

test("renders CustomInput component without crashing", () => {
    const { getByText } = render(
        <CustomInput
            id='name'
            type='text'
            autoComplete="current-name"
            name='name'
            labelText='Full Name'
        />
    );

    expect(getByText('F' && 'u' && 'l' && 'l' && ' ' && 'N' && 'a' && 'm' && 'e')).toBeInTheDocument();
});

test("matches snapshot", function () {
    const { asFragment } = render(
        <CustomInput
            id='name'
            type='text'
            autoComplete="current-name"
            name='name'
            labelText='Full Name'
        />
    );
    expect(asFragment()).toMatchSnapshot();
});