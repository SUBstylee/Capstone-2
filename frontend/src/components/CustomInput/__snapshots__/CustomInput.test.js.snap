// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="CustomInput"
  >
    <input
      autocomplete="current-name"
      id="name"
      name="name"
      type="text"
    />
    <label
      for="name"
    >
      <span
        style="transition-delay: 0ms;"
      >
        F
      </span>
      <span
        style="transition-delay: 50ms;"
      >
        u
      </span>
      <span
        style="transition-delay: 100ms;"
      >
        l
      </span>
      <span
        style="transition-delay: 150ms;"
      >
        l
      </span>
      <span
        style="transition-delay: 200ms;"
      >
         
      </span>
      <span
        style="transition-delay: 250ms;"
      >
        N
      </span>
      <span
        style="transition-delay: 300ms;"
      >
        a
      </span>
      <span
        style="transition-delay: 350ms;"
      >
        m
      </span>
      <span
        style="transition-delay: 400ms;"
      >
        e
      </span>
    </label>
  </div>
</DocumentFragment>
`;
